### PR TITLE
Explicit message in the logs when additional_includes file cannot be found

### PR DESCRIPTION
**Pull request by thomasp-ms**
When the additional_includes file cannot be found, log a message explaining the expected name. If a file with a wrong name is found, throw an exception.

#193

### CODE REVIEW

File: shrike/build/commands/prepare.py [hunk 1]
```diff
@@ -502,14 +502,26 @@ def component_is_active(self, component, modified_files) -> bool:
...
+        for modified_file in modified_files:
+            if self.is_in_subfolder(
+                modified_file, component
+            ) or self.is_in_additional_includes(
+                modified_file, component_additional_includes_contents
+            ):
+                return True
+        return False
+
+    def get_additional_includes_contents(self, component)-> Union[List[str], None]:
```
Comment (haozhestat) [inline shrike/build/commands/prepare.py:516]: Could you add a unit test for this new function `get_additional_includes_contents()`

**Review by haozhestat — CHANGES_REQUESTED** [review]

File: shrike/build/commands/prepare.py (hunk 1 above)
Comment (thomasp-ms) [reply shrike/build/commands/prepare.py:516]: Test added, even though it doesn't bring much (the only "new" thing about this function is the message in the logs; all the rest is the exact same logic without any change, but thrown into a function to make the code cleaner).

Comment (fuhuifang) [reply shrike/build/commands/prepare.py:516]: How about checking the log against the newly added message?

Comment (thomasp-ms) [reply shrike/build/commands/prepare.py:516]: Sure, I can add that too

Comment (thomasp-ms) [reply shrike/build/commands/prepare.py:516]: @haozhestat @fuhuifang does any of you have any clue why I'm running into an issue with one of the _register_ tests?

Comment (thomasp-ms) [reply shrike/build/commands/prepare.py:516]: At first I was getting errors about the `--label default` argument. After trying to remove it I'm now seeing things like "_ERROR: Unsupported component type: CommandComponent_" so I'm very confused...

Comment (haozhestat) [reply shrike/build/commands/prepare.py:516]: @thomasp-ms  and @fuhuifang : I guess that your build failed due to hitting the maximum number of component versions. I am doing some debugging now.

Comment (thomasp-ms) [reply shrike/build/commands/prepare.py:516]: thanks for the hotfix @haozhestat This PR is now ready for approval by you or @fuhuifang

**Comment by fuhuifang** [comment]
@thomasp-ms, could you bump up the version?

File: shrike/build/commands/prepare.py [hunk 2]
```diff
@@ -519,7 +531,8 @@ def component_is_active(self, component, modified_files) -> bool:
                 )
         else:
             component_additional_includes_contents = None
-        # make the paths in the additional_includes file absolute
+            log.info("No additional_includes file could be found for the component '" + component + "'. If you tried to create such a file, remember it should live next to the component spec file and should be named '{spec_file_name}.additional_includes'. For example, if the component spec file is named 'component_spec.yaml', the additional_includes file should be named 'component_spec.additional_includes'. In this specific case, the expected additional_includes file name is: '" + component_additional_includes_path + "'.")
```
Comment (dkmiller) [inline shrike/build/commands/prepare.py:534]: Two suggestions:

1. Break this comment into two lines.
2. Catch the case where there's a file `*.additional_includes`, where `*` is _not_ the component spec file name. We should warn "loudly" in that case.

**Review by dkmiller — CHANGES_REQUESTED** [review]

**Review by haozhestat — APPROVED** [review]

File: shrike/build/commands/prepare.py (hunk 2 above)
Comment (thomasp-ms) [reply shrike/build/commands/prepare.py]: @dkmiller Suggestions followed. I warned very loud by throwing an exception. Let me know if you meant just logging at a higher severity

**Review by dkmiller — APPROVED** [review]

**Pull request merged**